pandas/core/common.py

```diff
@@ -2016,14 +2016,15 @@ def _asarray_tuplesafe(values, dtype=None):
 
 
 def _index_labels_to_array(labels):
-    if isinstance(labels, (compat.string_types, tuple)):
+    from .index import Index # ABCIndex doesn't work here
```

Comment:
I *think* you might be able to eliminate this routine (``_index_labels_to_array``) entirely, its only in a couple of places

*and* I think this is overlapping a couple of other routines which do this (you'll have to look a bit in the code)